pkg/client/client.go

```diff
@@ -86,14 +86,18 @@ func NewDefaultClientWithTimeout(timeout time.Duration) (*Client, error) {
 	for {
 		select {
 		case <-timeoutAfter:
-			return c, fmt.Errorf("Failed to create cilium agent client after %f seconds timeout: %s", timeout.Seconds(), err)
+			return nil, fmt.Errorf("Failed to create cilium agent client after %f seconds timeout: %s", timeout.Seconds(), err)
 		default:
 		}
 
 		c, err = NewDefaultClient()
 		if err == nil {
-			break
+			_, err := c.Daemon.GetConfig(nil)
```

Comment:
Not-blocking: would be helpful to add a comment indicating that we make a request against the daemon here purely to see if it has connectivity (i.e., calling `GetConfig` vs. some other API call doesn't matter).